rlog.go

```diff
@@ -235,30 +250,143 @@ func (f filter) match(filename string, level int) (bool, bool) {
 }
 
+// updateIfNeeded returns a new value for an existing config item. The priority
+// flag indicates whether the new value should always override the old value.
+// Otherwise, the new value will not be used in case the old value is already
+// set.
+func updateIfNeeded(oldVal string, newVal string, priority bool) string {
+	if priority || oldVal == "" {
+		return newVal
+	} else {
+		return oldVal
+	}
+}
+
+// updateConfigFromFile reads a configuration from the specified config file.
+// It merges the supplied config with the new values.
+func updateConfigFromFile(config *rlogConfig) {
+	settingConfFile = config.confFile
+	// If no config file was specified we will default to a known location.
+	if settingConfFile == "" {
+		execName := filepath.Base(os.Args[0])
+		settingConfFile = fmt.Sprintf("/etc/rlog/%s.conf", execName)
+	}
+
+	// Scan over the config file, line by line
+	file, err := os.Open(settingConfFile)
+	defer file.Close()
+	if err != nil {
+		// Any error while attempting to open the logfile ignored. In many
+		// cases there won't even be a config file.
+		return
+	}
+
+	scanner := bufio.NewScanner(file)
+	i := 0
+	for scanner.Scan() {
+		i++
+		line := strings.TrimSpace(scanner.Text())
+		if line == "" {
+			continue
+		}
+		tokens := strings.SplitN(line, "=", 2)
+		if len(tokens) == 0 {
+			continue
+		}
+		if len(tokens) != 2 {
+			fmt.Fprintf(os.Stderr, "WARN: rlog - Malformed line in config file %s:%d\n",
+				settingConfFile, i)
+			fmt.Fprintf(os.Stderr, "%s\n", tokens)
+			continue
+		}
+		name := strings.TrimSpace(tokens[0])
+		val := strings.TrimSpace(tokens[1])
+
+		// If the name starts with a '!' then it should overwrite whatever we
+		// currently have in the config already.
+		priority := false
+		if name[0] == '!' {
+			priority = true
+			name = name[1:]
+		}
+
+		switch name {
+		case "RLOG_LOG_LEVEL":
+			config.logLevel = updateIfNeeded(config.logLevel, val, priority)
+		case "RLOG_TRACE_LEVEL":
+			config.traceLevel = updateIfNeeded(config.traceLevel, val, priority)
+		case "RLOG_TIME_FORMAT":
+			config.logTimeFormat = updateIfNeeded(config.logTimeFormat, val, priority)
+		case "RLOG_LOG_FILE":
+			config.logFile = updateIfNeeded(config.logFile, val, priority)
+		case "RLOG_CONF_FILE":
+			fmt.Fprintf(os.Stderr, "WARN: rlog - Ignoring attempt to change config file in config file %s:%d\n",
+				settingConfFile, i)
+		case "RLOG_LOG_STREAM":
+			val = strings.ToUpper(val)
+			config.logStream = updateIfNeeded(config.logStream, val, priority)
+		case "RLOG_LOG_NOTIME":
+			config.logNoTime = updateIfNeeded(config.logNoTime, val, priority)
+		case "RLOG_CALLER_INFO":
+			config.showCallerInfo = updateIfNeeded(config.showCallerInfo, val, priority)
+		case "RLOG_CONF_CHECK_INTERVAL":
+			config.confCheckInterv = updateIfNeeded(config.confCheckInterv, val, priority)
+		default:
+			fmt.Fprintf(os.Stderr, "WARN: rlog - Unknown setting name in config file %s:%d\n",
+				settingConfFile, i)
+			continue
```

Comment:
Indeed, it is. That's a left over from something else being below the switch statement in the past.